### PR TITLE
feat: support record events in redis

### PR DESCRIPTION
**Pull request by yashasvibajpai**
# Description

We want to support mirror mode for redis destination via the 'record' event type. Here, we are adding support for addition/updation & deletion based on action specified for the concerned record event.

## Linear Ticket

Resolves INT-1680

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **New Features**
	- Added support for insert, update, or delete actions in handling KV store events to ensure compatibility with different transformer versions.
	- Introduced a method for deleting keys from a Redis hash, enhancing data management capabilities.

### CODE REVIEW

File: router/customdestinationmanager/customdestinationmanager.go [hunk 1]
```diff
@@ -133,13 +133,25 @@ func (customManager *CustomManagerT) send(jsonData json.RawMessage, client inter
 	case KV:
 		var err error
 		kvManager, _ := client.(kvstoremanager.KVStoreManager)
-		// if the event supports HSET operation then use HSET
-		if kvstoremanager.IsHSETCompatibleEvent(jsonData) {
-			hash, key, value := kvstoremanager.ExtractHashKeyValueFromEvent(jsonData)
-			err = kvManager.HSet(hash, key, value)
+		// if the event type is record
+		if kvstoremanager.IsRecordCompatibleEvent(jsonData) {
```
Comment (saikumarrs) [inline router/customdestinationmanager/customdestinationmanager.go:137]: We should not bring the context of `record` here. Just check if the payload is compatible with delete.

File: router/customdestinationmanager/customdestinationmanager.go [hunk 2]
```diff
@@ -133,13 +133,25 @@ func (customManager *CustomManagerT) send(jsonData json.RawMessage, client inter
...
+			hash, key, value, action := kvstoremanager.ExtractHashKeyValueFromEvent(jsonData)
+			switch action {
+			case "insert":
+			case "update":
+				err = kvManager.HSet(hash, key, value)
```
Comment (saikumarrs) [inline router/customdestinationmanager/customdestinationmanager.go:142]: Don't we need `HMSet` also here?

**Review by koladilip — COMMENTED** [review]
Add test cases

File: services/kvstoremanager/redis.go [hunk 3]
```diff
@@ -150,3 +150,13 @@ func (m *redisManagerT) HSet(hash, key string, value interface{}) (err error) {
 	}
 	return err
 }
+
+func (m *redisManagerT) HDel(hash string, value ...string) (err error) {
+	ctx := context.Background()
+	if m.clusterMode {
+		_, err = m.clusterClient.HDel(ctx, hash, value...).Result()
```
[the remark this answers is not in the document]
Comment (koladilip) [reply services/kvstoremanager/redis.go:157]: Hash is correct but value should the fields which needs to be deleted from that hash, anyways this whole wrapper is written in very confusing way so we need to do clean up here

[the remark this answers is not in the document]
Comment (saikumarrs) [reply services/kvstoremanager/redis.go:157]: Is there a different API?

**Comment by yashasvibajpai** [comment]
@coderabbitai review

**Pull request closed without merging**